index.js

```diff
@@ -67,6 +67,15 @@ let cachedRecords;
 let credentials;
 
 /**
+ * Maps between passed in user name and internal user name.
+ *
+ * @type {Object}
+ *
+ * @private
+ */
+let usernameMap;
```

Comment:
I could change the name, but its intention isn't to cache usernames, but rather to map between the  username and the internal username. My concern is that naming it a cache could lead to confusion/misunderstanding down the line.